src/layer/tile/GridLayer.js

```diff
@@ -413,15 +413,19 @@ L.GridLayer = L.Layer.extend({
 		if (!map) { return; }
 
 		if (center === undefined) { center = map.getCenter(); }
-		if (zoom === undefined) { zoom = Math.round(map.getZoom()); }
+		if (zoom === undefined) { zoom = map.getZoom(); }
 
 		if (zoom > this.options.maxZoom ||
 			zoom < this.options.minZoom) { return; }
```

Comment:
The zoom level is rounded a few lines down; the original zoom level is needed for fetching the tile scale; and I didn't want to run more than one call to `map.getZoom()`.